packages/notification-managment-spa/src/app/form/form.component.ts

```diff
@@ -0,0 +1,131 @@
+import { Component, OnInit } from '@angular/core';
+import { FormsModule } from '@angular/forms';
+import { AppService } from '../app.service';
+import { UserProfile } from '../helper';
+
+@Component({
+  selector: 'op-form',
+  templateUrl: './form.component.html',
+  styleUrls: ['./form.component.scss']
+})
+export class FormComponent implements OnInit {
+  user = UserProfile;
+  days: Day[] = [
+    {
+      day: 'mon',
+      status: false,
+    },
+    {
+      day: 'tue',
+      status: false,
+    },
+    {
+      day: 'wed',
+      status: false,
+    },
+    {
+      day: 'thurs',
+      status: false,
+    },
+    {
+      day: 'fri',
+      status: false,
+    },
+    {
+      day: 'sat',
+      status: false,
+    },
+    {
+      day: 'sun',
+      status: false,
+    },
+  ];
+
+  events: EventType[] = [
+    {
+      name: 'create',
+      status: false,
+    },
+    {
+      name: 'update',
+      status: false,
+    },
+    {
+      name: 'delete',
+      status: false,
+    },
+  ];
+
+  applications = [
+    {
+      name: 'Outage Management',
+    },
+    {
+      name: 'TS Catalog',
+    },
+    {
+      name: 'Outage Management',
+    },
+  ];
+  channel: string;
+  type: string;
+  template: string;
+  schedule: Schedule;
+  createdBy: string;
+  createdOn: Date;
+  notificationFormData: NotificationSchema;
+  modalState: boolean;
+  toast: boolean;
+  state: boolean;
+
+  constructor(
+    private appService: AppService
+  ) { }
+
+  ngOnInit(): void {
+    this.stubFunction();
+  }
+
+  onSubmit(value) {
+    let option;
+    switch (value.trigger) {
+      case 'scheduled':
+        this.schedule = {
+          startDate: value.time,
+          repeats: this.days,
+        };
+        option = this.schedule;
+        break;
+      case 'triggered':
+        option = this.events;
+        break;
+      default:
+        option = null;
+    }
+    this.notificationFormData = {
+      template: this.template,
+      channel: value.channel,
+      type: value.trigger,
+      typeOption: option,
+      createdBy: this.createdBy,
+      createdOn: this.createdOn,
+      modifiedBy: null,
+      modifiedOn: null
+    };
+    this.appService.sendNotificationFormData(this.notificationFormData).subscribe(result => {
+      if (result) {
+        const pfeToast = window.document.querySelector('pfe-toast');
+        (pfeToast as any).open();
+      }
+    });
+
+  }
+  toggleModal() {
+    this.modalState = !this.modalState;
+  }
+
+  stubFunction() {
+    this.createdBy = '5de6432ad724996209eb9177';
```

Comment:
why this is statically listed.

Reply:
Removed this, as it was used for testing. When the microservice is ready it will automatically be populated